pkgs/development/python-modules/ocrmypdf/default.nix

```diff
@@ -0,0 +1,118 @@
+{ buildPythonPackage
+, cffi
+, chardet
+, fetchFromGitHub
+, ghostscript
+, hocr-tools
+, img2pdf
+, isPy3k
+, jbig2enc
+, leptonica
+, pdfminer
+, pillow
+, pikepdf
+, pngquant
+, pypdf2
+, pytest
+, pytest-helpers-namespace
+, pytest_xdist
+, pytestcov
+, pytestrunner
+, python-xmp-toolkit
+, qpdf
+, reportlab
+, ruffus
+, setuptools
+, setuptools-git
+, setuptools_scm
+, setuptools-scm-git-archive
+, stdenv
+, tesseract4
+, unpaper
+}:
+
+buildPythonPackage rec {
+  pname = "ocrmypdf";
+  version = "8.2.3";
+  disabled = ! isPy3k;
+
+  src = fetchFromGitHub {
+    owner = "jbarlow83";
+    repo = "OCRmyPDF";
+    rev = "v${version}";
+    sha256 = "1ldlyhxkav34y9d7g2kx3d4p26c2b82vnwi0ywnfynb16sav36d5";
+  };
+
+  buildInputs = [
+    pytest
+    pytestrunner
+    setuptools
+    setuptools_scm
+    setuptools-scm-git-archive
+  ];
+
+  propagatedBuildInputs = [
+    cffi
+    chardet
+    img2pdf
+    pdfminer
+    pikepdf
+    reportlab
+    ruffus
+  ] ++ runtimeDeps;
+
+  checkInputs = [
+    hocr-tools
+    pypdf2
+    pytest
+    pytest-helpers-namespace
+    pytest_xdist
+    pytestcov
+    pytestrunner
+    python-xmp-toolkit
+    setuptools
+  ] ++ runtimeDeps;
+
+  runtimeDeps = [
+    ghostscript
+    jbig2enc
+    leptonica
+    pngquant
+    qpdf
+    tesseract4
+    unpaper
+    pillow
+  ];
+
+  postPatch = ''
+    substituteInPlace src/ocrmypdf/leptonica.py \
+      --replace "ffi.dlopen(find_library('lept'))" \
+      'ffi.dlopen("${stdenv.lib.makeLibraryPath [leptonica]}/liblept${stdenv.hostPlatform.extensions.sharedLibrary}")'
+  '';
+
+  doCheck = false;
```

Comment:
Why are the tests disabled? Please include a comment.

Reply:
I enabled them. I had them disabled because they take over 20 minutes to run and I've probably built this over a hundred times ;_;

Reply:
20 minutes is a bit long. I think that's a good reason not it enable them.

Reply:
If you like I can disable them and put the comment. Should I do that instead?

Reply:
Unless there is a smaller test suite we should do that. I don't think 20 minutes is reasonable for a package like this.